armi/utils/tests/test_pathTools.py

```diff
@@ -70,6 +71,57 @@ def test_moduleAndAttributeExist(self):
         self.assertTrue(pathTools.moduleAndAttributeExist(thisFile + ":THIS_DIR"))
         self.assertTrue(pathTools.moduleAndAttributeExist(thisFile + ":PathToolsTests"))
 
+    def test_cleanPathOnValidPaths(self):
+        """Test cleanPaths by cleaning a path for a file, an empty folder, and a file containing a folder.
+        Test in an initialized armi context to include MPI."""
+        import armi
```

Comment:
This test doesn't run without this import?

Did you try?

Reply:
It would run, but it could not test MPI execution the way it is currently written.  It seems to me like there tend to be fewer integration-level tests rather than many unit tests, so I have written the test this way.  I'm happy to add unit tests that mock the global armi object and separately test the MPI execution path and non-MPI path.

Reply:
Hello @davis8dd !

1. GitHub Actions are not a good place for multi-threaded or multi-process code. And it's also not well-suited to integration tests.  So every project I've ever seen uses GitHub Actions more for unit tests.  This goes doubly so for big, scientific models whose integration tests might take days spread across many nodes in a huge computing cluster.
2. I definitely like the idea of having unit tests that test this method in the MPI and no-MPI pathways directly, even if we have to fake that a bit in a no-MPI unit test. Have you seen this class of unit tests, that does such testing?

https://github.com/terrapower/armi/blob/fd5d2dd31051206ce62696027f05a9304b499677/armi/tests/test_mpiActions.py#L30-L40

Reply:
Thank you for the feedback! I'll take another look at the test_mpiActions.py file and begin writing new tests.